os/arch/arm/src/imxrt/imxrt_hyperflash.c

```diff
@@ -478,12 +395,272 @@ static inline void flexspi_clock_init(void)
 	imxrt_clock_setdiv(kCLOCK_FlexspiDiv, 3);   /* flexspi clock 83M, DDR mode, internal clock 42M. */
 }
 
+/************************************************************************************
+ * Name: flexspi_clock_update
+ *
+ * Description:
+ *   Update clock setting in flexspi
+ *
+ ************************************************************************************/
+static inline void flexspi_clock_update(void)
+{
+	/* Program finished, speed the clock to 166M. */
+	imxrt_flexspi_enable(IMXRT_FLEXSPI, false);
+	imxrt_clock_disableclock(IMXRT_FLEXSPI_CLOCK);
+	imxrt_clock_setdiv(kCLOCK_FlexspiDiv, 0); /* flexspi clock 332M, DDR mode, internal clock 166M. */
+	imxrt_clock_enableclock(IMXRT_FLEXSPI_CLOCK);
+	imxrt_flexspi_enable(IMXRT_FLEXSPI, true);
+}
+
 /************************************************************************************
  * Public Functions
  ************************************************************************************/
 /****************************************************************************
  * Public Types
  ****************************************************************************/
+/************************************************************************************
+ * Name: imxrt_flexspi_nor_hyperbus_read
+ *
+ * Description:
+ *   Check if bus is busy and wait until exiting busy
+ *
+ ************************************************************************************/
+status_t imxrt_flexspi_nor_hyperbus_read(FLEXSPI_Type *base, uint32_t addr, uint32_t *buffer, uint32_t bytes)
+{
+	flexspi_transfer_t flashXfer;
+	status_t status;
+
+	flashXfer.deviceAddress = addr * 2;
+	flashXfer.port = kFLEXSPI_PortA1;
+	flashXfer.cmdType = kFLEXSPI_Read;
+	flashXfer.SeqNumber = 1;
+	flashXfer.seqIndex = HYPERFLASH_CMD_LUT_SEQ_IDX_READDATA;
+	flashXfer.data = buffer;
+	flashXfer.dataSize = bytes;
+	status = imxrt_flexspi_transferblocking(base, &flashXfer);
+
+	if (status != kStatus_Success) {
```

Comment:
If the status is not equal to success, you have to print a debug message and return negative value.